compiler/src/main/scala/dx/translator/cwl/ProcessTranslator.scala

```diff
@@ -575,14 +575,24 @@ case class ProcessTranslator(cwlBundle: CwlBundle,
         param.name -> param.dxType
       }.toMap
 
+      val (instanceType, container, attributes, requirements) = availableDependencies
```

Comment:
Please consider the following:
1. Creating an abstract class for `ProcessTranslator` and `CallableTranslator`.
2. Creating a common method for the line above. 
3. Not return tuples. They are hard to figure out what's happening and the return types.